examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -0,0 +1,235 @@
+        throw;
+
+    auto linesBuffer = driver->createDeviceLocalGPUBufferOnDedMem(triangleCount * 6 * sizeof(float));
+
+    const asset::SBufferBinding<video::IGPUBuffer> bufferBinding = { 0u ,linesBuffer };
+
+    //we can safely assume that all meshbuffers within mesh loaded from OBJ has same DS1 layout (used for camera-specific data)
+    //so we can create just one DS
+    asset::ICPUDescriptorSetLayout* ds1layout = mesh_raw->getMeshBuffer(0u)->getPipeline()->getLayout()->getDescriptorSetLayout(1u); //set 1u ---> 0u ?
+    uint32_t ds1UboBinding = 0u;
+    for (const auto& bnd : ds1layout->getBindings())
+        if (bnd.type==asset::EDT_UNIFORM_BUFFER)
+        {
+            ds1UboBinding = bnd.binding;
+            break;
+        }
+
+    size_t neededDS1UBOsz = 0ull;
+    {
+        auto pipelineMetadata = static_cast<const asset::IPipelineMetadata*>(mesh_raw->getMeshBuffer(0u)->getPipeline()->getMetadata());
+        for (const auto& shdrIn : pipelineMetadata->getCommonRequiredInputs())
+            if (shdrIn.descriptorSection.type==asset::IPipelineMetadata::ShaderInput::ET_UNIFORM_BUFFER && shdrIn.descriptorSection.uniformBufferObject.set==1u && shdrIn.descriptorSection.uniformBufferObject.binding==ds1UboBinding)
+                neededDS1UBOsz = std::max<size_t>(neededDS1UBOsz, shdrIn.descriptorSection.uniformBufferObject.relByteoffset+shdrIn.descriptorSection.uniformBufferObject.bytesize);
+    }
+
+    auto gpuds1layout = driver->getGPUObjectsFromAssets(&ds1layout, &ds1layout+1)->front();
+
+    auto gpuubo = driver->createDeviceLocalGPUBufferOnDedMem(neededDS1UBOsz);
+    auto gpuds1 = driver->createGPUDescriptorSet(std::move(gpuds1layout));
+    {
+        video::IGPUDescriptorSet::SWriteDescriptorSet write;
+        write.dstSet = gpuds1.get();
+        write.binding = ds1UboBinding;
+        write.count = 1u;
+        write.arrayElement = 0u;
+        write.descriptorType = asset::EDT_UNIFORM_BUFFER;
+        video::IGPUDescriptorSet::SDescriptorInfo info;
+        {
+            info.desc = gpuubo;
+            info.buffer.offset = 0ull;
+            info.buffer.size = neededDS1UBOsz;
+        }
+        write.info = &info;
+        driver->updateDescriptorSets(1u, &write, 0u, nullptr);
+    }
+
+    auto gpumesh = driver->getGPUObjectsFromAssets(&mesh_raw, &mesh_raw+1)->front();
+
+	//! we want to move around the scene and view it from different angles
+	scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeFPS(0,100.0f,0.5f);
+
+	camera->setPosition(core::vector3df(-4,0,0));
+	camera->setTarget(core::vector3df(0,0,0));
+	camera->setNearValue(1.f);
+	camera->setFarValue(5000.0f);
+
+    smgr->setActiveCamera(camera);
+
+	uint64_t lastFPSTime = 0;
+	while(device->run() && receiver.keepOpen())
+	{
+		driver->beginScene(true, true, video::SColor(255,255,255,255) );
+        
+        // zero out buffer LineCount
+        driver->fillBuffer(lineCountBuffer.get(),0,sizeof(uint32_t),0);
```

Comment:
you should compute the `offsetof(DrawIndirectParameters_t,count)` and only zero-out the count